### PR TITLE
Mergeable edit to include test suite success for service source

### PR DESCRIPTION
**Pull request by danthegoodman1**

### CODE REVIEW

**Comment by danthegoodman1** [comment]
@balena-ci rebase

**Comment by danthegoodman1** [comment]
@Hades32 squashed to 1 commit and ready to go

File: lib/cards/service-source.ts [hunk 1]
```diff
@@ -26,7 +26,17 @@ export const serviceSource: ContractDefinition = {
...
 									description: 'all platforms have an image',
 									type: 'boolean',
-									$$formula: 'EVERY(VALUES(contract.data.platforms), "image")',
+									$$formula: `
+									EVERY(VALUES(contract.data.platforms), "image") &&
+									contract.data.$transformer.backflow &&
+									contract.data.$transformer.backflow.filter(
+										function(b){
+											return b &&
+														 b.type.startsWith("t-product-os-test-run") &&
```
Comment (Hades32) [inline lib/cards/service-source.ts:35]:
```suggestion
														 b.type.startsWith("type-product-os-t-test-run@") &&
```

**Review by Hades32 — CHANGES_REQUESTED** [review]
basically LGTM, but I just remembered that we said, we want to move this type to its own repo as otherwise a type from a plugin would reference an "external" type (i.e. `type-product-os-t-test-run`). This should also make rolling out this change much less "dangerous".

**Comment by danthegoodman1** [comment]
@balena-ci rebase

**Comment by danthegoodman1** [comment]
> basically LGTM, but I just remembered that we said, we want to move this type to its own repo as otherwise a type from a plugin would reference an "external" type (i.e. `type-product-os-t-test-run`). This should also make rolling out this change much less "dangerous".
> 
> If you do that, you can also migrate `type-product-os-t-test-run` to use `type-source`

I thought that was to be in a separate PR, my mistake, I'll get that handled right away.

**Comment by danthegoodman1** [comment]
@balena-ci rebase

**Comment by danthegoodman1** [comment]
See https://github.com/product-os/t-service-source/pull/1

**Pull request closed without merging**